src/Simple.OData.Client.Core/ODataRequest.cs

```diff
@@ -57,7 +58,13 @@ public string[] Accept
         public bool ReturnsScalarResult { get; set; }
         public bool ResultRequired { get; set; }
         public bool CheckOptimisticConcurrency { get; set; }
-        public readonly IDictionary<string, string> Headers = new Dictionary<string, string>();
+        public IDictionary<string, string> Headers { get; } = new Dictionary<string, string>();
+
+        /// <summary>
+        /// Internal state used for batch requests. Needed in order to replay operations to assign return values to closures.
+        /// </summary>
+        internal IList<int> BatchResponseIndeces { get; set; } = new List<int>();
+        internal IList<Func<IODataClient, Task>> BatchActions { get; set; } = new List<Func<IODataClient, Task>>();
```

Comment:
Why was this needed as part of this work? (curious)

If its a perf gain, it would be better in another PR.

Reply:
This is a fix already commited and merge, I will remove it from this PR.

Reply:
The intent behind the seemlingly unrelated functional changes was the ability to test the implementation. For all the singular operations you can get the request using the GetRequestFor() API. There is no such API for batch and digging deeper I found out that it's impossible to return an ODataRequest and then send the request without breaking the feature with the returning closures. I think it's much more sound architecturally, to encapsulate this "state" of actions and indeces inside the request object and pass it on for execution. So I can revert these changes and just commit the header setting part but there will simply be no way to write an end to end test for this feature.

Reply:
I'm not familiar enough with this pathway to know how big of an impact this change is, perhaps @object  or @phatcher  can weigh in.

It seems reasonable on face value (with the minor nit to change `Indeces` to `Indices` or `Indexes`).

Reply:
Indeces is a type on my part, it was supposed to be Indices. It's minor I can revert to indexes it's internal so it doesn't break anything.